repos/fdbt-site/src/pages/products/services.tsx

```diff
@@ -187,7 +187,18 @@ export const matchProductsToServices = (
 export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: ServicesProps }> => {
     const noc = getAndValidateNoc(ctx);
     const dataSource = !!getSessionAttribute(ctx.req, MULTI_MODAL_ATTRIBUTE) ? 'tnds' : 'bods';
-    const services: MyFaresService[] = await getBodsOrTndsServicesByNoc(noc, dataSource);
+    const allServices: MyFaresService[] = await getBodsOrTndsServicesByNoc(noc, dataSource);
+    const uniqueServiceList: string[] = [];
+    const services: MyFaresService[] = [];
+
+    allServices.forEach((service) => {
+        const service_name = `${service.lineId}#${service.startDate}#${service.endDate}`;
+        if (!uniqueServiceList.includes(service_name)) {
```

Comment:
service_name is a python naming standard, we use camelCase in JS/TS.

We need a function which takes an array of services as an argument, and returns an array of services, but removes any duplicates (any services that share the same lineId, startDate and endDate).

Then we need to write unit tests for this function, with different scenarios.

Reply:
Then the usage on this page will be 

` const services: MyFaresService[] = await getBodsOrTndsServicesByNoc(noc, dataSource);`
`const servicesWithNoDuplicates = removeDuplicateServices(services);`